### PR TITLE
Issue 1208

### PR DESCRIPTION
**Pull request by AbfrmBlr**
Not to be merged yet

### CODE REVIEW

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java [hunk 1]
```diff
@@ -247,7 +249,31 @@ public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEn
 
   @Override
   public void handleReconnect(ClientDescriptor clientDescriptor, byte[] extendedReconnectData) {
-    //nothing to do
+    ClientState clientState = this.clientStateMap.get(clientDescriptor);
+    if (clientState == null) {
+      LOGGER.error("Client {} trying to reconnect is not connected to entity.", clientDescriptor);
+      return;
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: I would throw an AssertionError here - this must be impossible from the platform's perspective!

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java [hunk 2]
```diff
@@ -247,7 +249,31 @@ public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEn
...
+    }
+    clientState.attach();
+    Set<String> cacheIds = reconnectDataCodec.decode(extendedReconnectData);
+    for (final String cacheId : cacheIds) {
+      ServerStoreImpl serverStore = ehcacheStateService.getStore(cacheId);
+      if (serverStore == null) {
+        //Client removes the cache's reference only when destroy has successfully completed
+        //This happens only when client thinks destroy is still not complete
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: I do not understand these comments

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcachePassiveEntity.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        default:
+          throw new IllegalMessageException("Unknown message : " + message);
+      }
+    } catch (Exception e) {
+      LOGGER.error("Unexpected exception raised during operation: " + message, e);
+    }
+
+  }
+
+  private void invokeLifeCycleOperation(LifecycleMessage message) throws ClusterException{
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcachePassiveEntity.java]: I have the impression that `invoke` and this method could be pulled into a super abstract type that passive and active version of the entity would implement.

Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcachePassiveEntity.java]: Forget what I said ... unless we hide the difference in parameters between a message with or without a `ClientDescriptor`

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java (hunk 2 above)
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: Got precision from @AbfrmBlr - given that, the following log statement should be `warn` level.

**Comment by AbfrmBlr** [comment]
It will good if this can be merged. This does not close #1208. This is a partial PR, following PRs shall contain message de-duplication and refactoring around passthrough replication testing in `UnitTestConnectionService`.

**Pull request merged**